bundles/org.openhab.ui/web/src/assets/definitions/blockly/ohblocks_timers.js

```diff
@@ -0,0 +1,326 @@
+      this.setNextStatement(true, null)
+      this.setColour(230)
+      this.setTooltip('Simple Timer creation without further control of retriggering, canceling or the like')
+      this.setHelpUrl('https://www.openhab.org/docs/configuration/actions.html#timers')
+    }
+  }
+
+  /*
+  * Simple Timer creation without further control of retriggering, canceling or the like
+  * Code part
+  */
+  Blockly.JavaScript['oh_simpleTimer'] = function (block) {
+    addScriptExecution()
+    addZonedDateTime()
+    let numberDelay = block.getFieldValue('delay')
+    let dropdownTimeUnit = block.getFieldValue('time_unit')
+    let statementsTimer = Blockly.JavaScript.statementToCode(block, 'timer')
+    let unitsFunction = ''
+    switch (dropdownTimeUnit) {
+      case 'seconds' :
+        unitsFunction = 'plusSeconds'
+        break
+      case 'minutes' :
+        unitsFunction = 'plusMinutes'
+        break
+      case 'hours' :
+        unitsFunction = 'plusHours'
+        break
+    }
+    let code = `scriptExecution.createTimer(zonedDateTime.now().${unitsFunction}(${numberDelay}), function(){\n`
+    code += statementsTimer
+    code += '})\n'
+    return code
+  }
+
+  /*
+  * Creates a named timer that starts after the defined delay provided my the number and the unit of time
+  * Blockly part
+  */
+  Blockly.Blocks['oh_namedTimer'] = {
+    init: function () {
+      this.appendStatementInput('runMe')
+        .setCheck(null)
+        .appendField('Run This')
+      this.appendValueInput('delay')
+        .setCheck(null)
+        .appendField('After')
+        .appendField(new Blockly.FieldDropdown([['seconds', 'plusSeconds'], ['minutes', 'plusMinutes'], ['hours', 'plusHours'], ['days', 'plusDays'], ['weeks', 'plusWeeks'], ['months', 'plusMonths']]), 'delayUnits')
+      this.appendValueInput('timerName')
+        .setCheck(null)
+        .appendField('with Timer named')
+      this.setColour(230)
+      this.setPreviousStatement(true, null)
+      this.setNextStatement(true, null)
+      this.setTooltip('Create a named timer')
+      this.setHelpUrl('https://www.openhab.org/docs/configuration/actions.html#timers')
+    }
+  }
+
+  /*
+  * Creates a named timer that starts after the defined delay provided my the number and the unit of time
+  * Code part
+  */
+  Blockly.JavaScript['oh_namedTimer'] = function (block) {
+    addScriptExecution()
+    addZonedDateTime()
+
+    let runme = Blockly.JavaScript.statementToCode(block, 'runMe')
+    let delayunits = block.getFieldValue('delayUnits')
+    let delay = Blockly.JavaScript.valueToCode(block, 'delay', Blockly.JavaScript.ORDER_ATOMIC)
+    let timerName = Blockly.JavaScript.valueToCode(block, 'timerName', Blockly.JavaScript.ORDER_ATOMIC).replace(/'/g, '')
+    addGlobalVar(timerName)
+
+    let code = `if (this.${timerName} === null) {\n`
+    code += `\tthis.${timerName} = scriptExecution.createTimer(zonedDateTime.now().${delayunits}(${delay}), function(){\n`
```

Comment:
I believe Blockly indents code with 4 spaces, not tabs?

Reply:
Also I don't like having global variables defined for every timer, would it be possible to put them all under an object, i.e. `this.timers.${timerName}`?
(Naturally this object would have to be initialized and checked for nullness in other blocks).

Reply:
Regarding the indenting I am not sure about tabs or 4 spaces. I can do that if you want (tabs though are nicer in the code I think). However, keep in mind, the indenting is not perfect, in particular in multi-lines and code within code. It is better than nothing but don't expect perfectly generated readable code (at least I have not found any hints during research how to do that in a better way).